spec/unit/provider/oneview_enclosure_synergy_spec.rb

```diff
@@ -0,0 +1,154 @@
+
+  let(:test) { resourcetype.new(@client, resource['data']) }
+
+  let(:resource) do
+    Puppet::Type.type(:oneview_enclosure).new(
+      name: 'Enclosure',
+      ensure: 'present',
+      data:
+          {
+            'name' => 'Puppet_Test_Enclosure',
+            'hostname' => '172.18.1.13',
+            'username' => 'dcs',
+            'password' => 'dcs',
+            'enclosureGroupUri' => '/rest/',
+            'licensingIntent' => 'OneView',
+            'refreshState' => 'RefreshPending',
+            'utilization_parameters' =>
+            {
+              'view' => 'day'
+            }
+          },
+      provider: 'synergy'
+    )
+  end
+
+  context 'given the min parameters' do
+    before(:each) do
+      allow(resourcetype).to receive(:find_by).with(anything, resource['data']).and_return([test])
+      provider.exists?
+    end
+
+    it 'should be an instance of the provider synergy' do
+      expect(provider).to be_an_instance_of Puppet::Type.type(:oneview_enclosure).provider(:synergy)
+    end
+
+    it 'should be able to get the environmental configuration' do
+      allow_any_instance_of(resourcetype).to receive(:environmental_configuration).and_return('Test')
+      expect(provider.get_environmental_configuration).to be
+    end
+
+    it 'should be able to set the configuration' do
+      allow_any_instance_of(resourcetype).to receive(:configuration).and_return('Test')
+      expect(provider.set_configuration).to be
+    end
+
+    it 'should be able to set the refresh state' do
+      allow_any_instance_of(resourcetype).to receive(:set_refresh_state).and_return('Test')
+      expect(provider.set_refresh_state).to be
+    end
+
+    it 'should be able to set the refresh state' do
+      allow_any_instance_of(resourcetype).to receive(:utilization).with(resource['data']['utilization_parameters']).and_return('Test')
+      expect(provider.get_utilization).to be
+    end
+
+    it 'should be able to set the refresh state' do
+      allow_any_instance_of(resourcetype).to receive(:script).and_return('Test')
+      expect(provider.get_script).to be
+    end
+
+    it 'should be able to set the refresh state' do
+      expect { provider.get_single_sign_on }.to raise_error(RuntimeError)
+    end
+
+    it 'should be able to work specifying a name instead of an uri' do
+      resource['data']['enclosureGroupUri'] = 'Test'
+      test = resourcetype.new(@client, resource['data'])
+      allow(OneviewSDK::EnclosureGroup).to receive(:find_by).with(anything, name: resource['data']['enclosureGroupUri']).and_return([test])
+      allow(resourcetype).to receive(:find_by).with(anything, resource['data']).and_return([test])
+      expect(provider.exists?).to be
+    end
+
+    it 'deletes the resource' do
+      resource['data']['uri'] = '/rest/fake'
+      test = resourcetype.new(@client, resource['data'])
+      allow(resourcetype).to receive(:find_by).with(anything, resource['data']).and_return([test])
```

Comment:
no need for multiple find_bys for different inputs and same return, just leave one without `with`